src/metabase/models/notification.clj

```diff
@@ -370,6 +372,71 @@
           :send_once      false}
          instance))
 
+;; ------------------------------------------------------------------------------------------------;;
+;;                                         Permissions                                             ;;
+;; ------------------------------------------------------------------------------------------------;;
+
+(defn current-user-can-read-payload?
+  "Check if the current user can read the payload of a notification."
+  [notification]
+  (case (:payload_type notification)
+    :notification/card
+    (mi/can-read? :model/Card (-> notification :payload :card_id))
+
+    :notification/system-event
+    (mi/superuser?)
+
+    :notification/testing
+    true))
+
+(defn current-user-is-recipient?
+  "Check if the current user is a recipient of a notification."
+  [notification]
+  (->> (:handlers (t2/hydrate notification [:handlers :recipients]))
+       (mapcat :recipients)
+       (map :user_id)
+       set
+       (some #{(mi/current-user-id)})
+       boolean))
+
+(defn current-user-is-creator?
+  "Check if the current user is the creator of a notification."
+  [notification]
+  (= (:creator_id notification) (mi/current-user-id)))
+
+(defmethod mi/can-read? :model/Notification
+  ([notification]
+   ;; users can view a notification if they can view the payload or if they are the creator
+   (or
+    (mi/superuser?)
+    (current-user-is-creator? notification)
+    (current-user-is-recipient? notification)))
+  ([_ pk]
+   (mi/can-read? (t2/select-one :model/Notification pk))))
+
+(defmethod mi/can-create? :model/Notification
+  [_ notification]
+  (or
+   (mi/superuser?)
+   (current-user-can-read-payload? notification)
+   ;; if ee is enabled, prevent users without subscription permissions from creating notifications
```

Comment:
this comment is confusing - this clause is additive, not restrictive

Reply:
hmm, it was wrong before!

lemme try again:
- when advanced permissions is disabled: only superuser or user can read payload can create
- when advanced permissions is enabled: only superuser or (user that can create payload AND has subscriptions permission) can create

853e3118dd

Reply:
that makes more sense!